### PR TITLE
standalone evaluator

### PR DESCRIPTION
**Pull request by leokraemer**

### CODE REVIEW

File: evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/GoldEntry.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    if (!getOldId().equals(goldEntry.getOldId())) return false;
+    if (!getFid().equals(goldEntry.getFid())) return false;
+    if (!getMathInputTex().equals(goldEntry.getMathInputTex())) return false;
+    if (!getTitle().equals(goldEntry.getTitle())) return false;
+    return getDefinitions().equals(goldEntry.getDefinitions());
+
+  }
+
+  @Override
+  public int hashCode() {
```
Comment (physikerwelt) [inline evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/GoldEntry.java]: can you explain, why you need a custom hash function?

File: evaluation/src/test/java/com/formulasearchengine/mlp/evaluation/ExtractionTests.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  public static final String GOLDFILE_SAMPLE = FOLDER + "gold_sample.json";
+  public static final String GOLDFILE_SAMPLE_2 = FOLDER + "gold_sample2.json";
+  public static final String GOLDFILE_UNSORTED = FOLDER + "gold_unsorted_sample.json";
+  public static final String EXTRACTIONS = FOLDER + "extraction.csv";
+  public static final String EXTRACTIONS_SAMPLE = FOLDER + "extraction_sample.csv";
+  public static final String EXTRACTIONS_UNSORTED = FOLDER + "extraction_unsorted.csv";
+  public static final String EXTRACTIONS_SAMPLE_WRONG = FOLDER + "extraction_sample_wrong.csv";
+  public static final String EXTRACTIONS_SAMPLE_WRONG_2 = FOLDER + "extraction_sample_wrong2.csv";
+  public static final String EXTRACTIONS_SAMPLE_WIKIDATA = FOLDER + "extraction_sample_wikidata.csv";
+  public static final String EXTRACTIONS_SAMPLE_WIKIDATA_2 = FOLDER + "extraction_sample_wikidata2.csv";
```
Comment (physikerwelt) [inline evaluation/src/test/java/com/formulasearchengine/mlp/evaluation/ExtractionTests.java]: 👍

File: evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/GoldEntry.java (hunk 1 above)
Comment (leokraemer) [inline evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/GoldEntry.java]: I needed the equals to make the list comparison for the `definitions` work as expected, this implies a custom hashCode. Also it is the one generated by Idea.

File: evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/Evaluator.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+  /**
+   * Evaluate an extraction result against the mlp gold standard.
+   * @param extractions multimap containing the identifiers and definitions for every formula.
+   * @param gold the gold data {@link #readGoldEntries}
+   * @return [true positives, false negatives, false positives]
+   */
+  public int[] evaluate(Multimap<String, IdentifierDefinition> extractions, List<GoldEntry> gold) {
+    int totalNumberOfIdentifiers = (int) gold.stream().flatMap(ge -> ge.getDefinitions().stream().map(i -> i.getIdentifier()).distinct()).count();
+    //initialize [true positives, false negatives, false positives, number of wikidata links matched] array
```
Comment (leokraemer) [inline evaluation/src/main/java/com/formulasearchengine/mlp/evaluation/Evaluator.java]: What code?

**Pull request merged**